### PR TITLE
Switch tar verification to a manifest comparison.

### PR DESCRIPTION
**Pull request by jonmeow**
This removes the install marker-relative path checks, and replaces it with bidirectional verification: previously, files in the tar file had to be in install data, but there was no check that files in install data were all in the tar.

### CODE REVIEW

File: toolchain/install/toolchain_tar_test.py [hunk 1]
```diff
@@ -22,42 +32,42 @@ def main() -> None:
...
-            sys.exit(
-                "ERROR: File `{0}` is not in tar file.".format(relative_path)
-            )
+    # Verify file sets match.
+    tar_okay = expect_empty_set(args.tar_file, tar_files - install_files)
+    install_okay = expect_empty_set(
+        args.install_data_manifest, install_files - tar_files
+    )
+    if not (tar_okay and install_okay):
+        exit("error: tar and install data did not match.")
```
Comment (chandlerc) [inline toolchain/install/toolchain_tar_test.py]: Is there a trivial function to print the difference here? Not worth lots of code, mostly wondering if Python already has this.

**Review by chandlerc — APPROVED** [review]
Very nice! Glad to see the TODO go away. Totally optional comment inline.

File: toolchain/install/toolchain_tar_test.py (hunk 1 above)
Comment (jonmeow) [reply toolchain/install/toolchain_tar_test.py]: Can you please clarify what you're looking for versus what `expect_empty_set` is printing?

Comment (chandlerc) [reply toolchain/install/toolchain_tar_test.py]: Sorry, I missed that `expect_empty_set` actually printed it out above, for some reason I thought it was just verifying empty with an expect framework. Anyways, change still LG.

**Review by chandlerc — APPROVED** [review]

File: toolchain/install/toolchain_tar_test.py (hunk 1 above)
Comment (jonmeow) [reply toolchain/install/toolchain_tar_test.py]: Thinking about this further, switched to `unittest` which has better support for this, with less code. To give you an idea:

```
======================================================================
FAIL: test_tar (__main__.ToolchainTarTest.test_tar)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/jperkins/.cache/bazel/_bazel_jperkins/85deb7d9d96f7e0e80b42618a55969d7/sandbox/linux-sandbox/10463/execroot/_main/bazel-out/k8-fastbuild/bin/toolchain/install/carbon_toolchain_tar_test.runfiles/_main/toolchain/install/toolchain_tar_test.py", line 47, in test_tar
    self.assertSetEqual(install_files, tar_files)
AssertionError: Items in the first set but not the second:
'lib/carbon/carbon_install.txt'
'lib/carbon/carbon-busybox'
'lib/carbon/llvm/bin/lld'
Items in the second set but not the first:
'lld'
'carbon-busybox'
'carbon_install.txt'

----------------------------------------------------------------------
```

Comment (chandlerc) [reply toolchain/install/toolchain_tar_test.py]: Wow, very nice! Thanks!

**Review by chandlerc — APPROVED** [review]
(and LG with new unittest module based code)

**Pull request merged**